src/Umbraco.Web/Trees/ContentTreeController.cs

```diff
@@ -228,25 +225,25 @@ protected override IEnumerable<IEntitySlim> GetChildEntities(string id, FormData
         protected MenuItemCollection GetAllNodeMenuItems(IUmbracoEntity item)
         {
             var menu = new MenuItemCollection();
-            AddActionNode<ActionNew>(item, menu);
-            AddActionNode<ActionDelete>(item, menu);
-
-            AddActionNode<ActionCreateBlueprintFromContent>(item, menu);
-
-            //need to ensure some of these are converted to the legacy system - until we upgrade them all to be angularized.
-            AddActionNode<ActionMove>(item, menu, true);
-            AddActionNode<ActionCopy>(item, menu);
-
+            AddActionNode<ActionNew>(item, menu, opensDialog: true);
+            AddActionNode<ActionDelete>(item, menu, opensDialog: true);
+            AddActionNode<ActionCreateBlueprintFromContent>(item, menu, opensDialog: true);
+            AddActionNode<ActionMove>(item, menu, true, opensDialog: true);
+            AddActionNode<ActionCopy>(item, menu, opensDialog: true);
             AddActionNode<ActionSort>(item, menu, true);
+            AddActionNode<ActionAssignDomain>(item, menu, opensDialog: true);
+            AddActionNode<ActionRights>(item, menu, opensDialog: true);
+            //fixme - conver this editor to angular
+            AddActionNode<ActionProtect>(item, menu, true, convert: true, opensDialog: true);
 
-            AddActionNode<ActionToPublish>(item, menu, convert: true);
-            AddActionNode<ActionAssignDomain>(item, menu);
-            AddActionNode<ActionRights>(item, menu, convert: true);
-            AddActionNode<ActionProtect>(item, menu, true, true);
-            
-            AddActionNode<ActionNotify>(item, menu, true);
+            menu.Items.Add(new MenuItem("notify", Services.TextService)
```

Comment:
Why is this IAction not added to the menu item list with a Type and use the new MenuItem ctor?
Its not a big deal, just more to do with consistency with all the others been declared the other way

Reply:
Because there is no `IAction` for notify. `IAction` is for only 2 things: assigning permissions and/or assigning notifications. `IAction` doesn't define menu items, but we have helper methods to be able to convert an `IAction` to a `MenuItem`. Tree's only deal with `MenuItems` after all. Now you if create an `IAction` that is not permission assignable or notification assignable you are going to get an exception because `IAction` is **not** for building menu items. Again, to create a menu item from an `IAction` is simply syntactic sugar.